extensions/replication/tasks/ReplicateObject.js

```diff
@@ -359,7 +362,23 @@ class ReplicateObject extends BackbeatTask {
                 return doneOnce(err);
             }
             partObj.setDataLocation(data.Location[0]);
-            return doneOnce(null, partObj.getValue());
+            return async.series([
+                next => {
+                    if (partNumber === 1) {
+                        const extMetrics = getExtMetrics(this.site,
```

Comment:
Maybe you should consider doing this at the beginning of the process rather than after having put the first part (if possible), so you have the total available as soon as the object starts being put, instead of after the first part is complete. Note that here you can also have some completion metrics written for other parts even before the "queued" key is written, e.g. if part #2 and #3 are completed but not #1 yet. Not sure this has an important impact but just want to point it out, and pushing the "queued" key first would avoid this also.

Reply:
Very good point. I've updated this.